perf_test: fix fio offset_increment

iomartin opened the pull request:
The 100M offset_increment was not enough to prevent overlap, set it to
the same value as "size" to prevent threads from overlapping.

While we're at it, also make it aligned to stripe size.

dlsloan commented on perf_test:
```diff
@@ -29,17 +29,19 @@ def iops_suffix(val):
     }
 
     rw = "write" if write else "read"
+    offset_increment = fio.args["size"] + stripe_size
```
FIO args accepts suffix strings, need a conversion function here

dlsloan replied:
this also doesn't align the offset_increment to the stripe_size, should be (size + stripe_size - 1) // stripe_size * stripe_size

iomartin replied:
> FIO args accepts suffix strings, need a conversion function here

the `size` is a hardcoded constant (for now at least). And `stripe_size` is computed from `chunk_size`, which is already converted [during the arg parsing](https://github.com/Eideticom/raid5-tests/blob/cecded61d38276c0d08b66b2c286994ac25318b0/perf_test#L99).

> this also doesn't align the offset_increment to the stripe_size, should be (size + stripe_size - 1) // stripe_size * stripe_size

Hmm right, will fix

dlsloan replied:
But they are a set of args passed directly into fio, I'm assuming this is based off of the noloadfs one and adding an fio arg is as simple as adding an FIO arg.  The expectation should be fio's expectation so we don't need to keep track of what's special and what's just an fio arg.

iomartin replied:
Yeah, this is based on the noloadfs fio.py. But note that this offset_increment computation is done in the perf_test script, not on fio.py, and in the perf_test it's not possible to change the `size` argument.

Although I supposed if we changed the base fio.py then this would break... So, sure, I'll add a suffix to bytes conversion

lsgunth replied:
Yeah, this is definitely confusing but I think the problem is with the original code. We have self.args which looks to almost entirely be a compile time constant even though it is created in __init__.... then we add self.SIZE which is a compile time constant and repeated in self.args; -- also a constant.

And thus it's not clear how to treat args at all.... David claims we have to impose fios rules on it, though why that's the case isn't clear. Worse yet we have something called args that doesn't seem like a sensible inter-module interface being used from another module.... 

Just because you have a variable with no underscores and nobody indicated it was private doesn't mean it's a good idea to just treat it as public interface. That's how you create a giant ball of spaghetti. Nobody reading the Fio code would think that args["size"] is a public interface with format promises to some other module.

If the calling code cares about the size, then it should probably be passing that size to the Fio class. Not having Fio specify it as a constant in a poorly named array and then have the calling code grab it from the class's variables.

iomartin committed:
perf_test: allow to set perf size

iomartin committed:
perf_test: align offset_increment

Make the fio IO non overlapping between threads and aligned to stripe
size to favour the fast path.

iomartin committed:
fio.py: increase offset_increment

Make it the same as "size" so that threads don't overlap.

iomartin commented:
Thanks for the suggestions. I reworked this a bit.

`perf_test` now takes a `--fio-size` argument and we use that to compute the `offset_increment`.

dlsloan reviewed (approved):
works for me

lsgunth reviewed (approved):
Looks better to me.
